WorkflowSwiftUI/Sources/Derived/ObservableState.swift

```diff
@@ -113,6 +113,20 @@ public func _$isIdentityEqual<T: ObservableState>(
     lhs._$id == rhs._$id
 }
 
+@inlinable
+public func _$isIdentityEqual<T: ObservableState>(
```

Comment:
This addition is not strictly needed for scoping, but makes the optional case much more efficient when going from nil to nil.

Reply:
Is it worth moving this to a file outside of the `Derived` folder? It'd probably make diffs between this and the TCA ObservableState easier.

Reply:
🤷  previously [I was asked](https://github.com/square/workflow-swift/pull/283#discussion_r1681291606) to put it _in_ this folder for the same reason.

Reply:
I just mean this function specifically (if it's not from the upstream repo)